samcli/lib/package/artifact_exporter.py

```diff
@@ -200,7 +200,11 @@ def make_zip(file_name, source_root):
                 for filename in files:
                     full_path = os.path.join(root, filename)
                     relative_path = os.path.relpath(full_path, source_root)
-                    zf.write(full_path, relative_path)
+                    with open(filename, 'rb') as data:
+                        file_bytes = data.read()
+                        info = zipfile.ZipInfo(relative_path)
+                        info.external_attr = 0o777 << 16L
```

Comment:
Can we scope this to only files that actually need it? We should not be mutating all file permissions in the zip, seems like an escalation of privileged.

Reply:
I'll dive a bit deeper and check if we can extract the permission bits using python standard library and match it exactly. Failing that, I'll only try to match this for executable files.

Reply:
We should also scope this to only windows. From my understanding, this is really only a problem on windows because you can't set linux like permissions.